### PR TITLE
simulationio: New package

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import os
+
+class Simulationio(CMakePackage):
+    """SimulationIO: Efficient and convenient I/O for large PDE simulations"""
+    homepage = "https://github.com/eschnett/SimulationIO"
+    url= "https://github.com/eschnett/SimulationIO/archive/version/0.1.0.tar.gz"
+
+    version('1.0.0', '5cbf1d0084eb436d861ffcdd297eaa08')
+    version('0.1.0', '00f7dabc08ed1ab77858785ce0809f50')
+    version('master',
```
Comment (davydden) [inline var/spack/repos/builtin/packages/simulationio/package.py]: develop

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        pythonpath = (
+            glob.glob(join_path(spec['py-h5py'].prefix, "lib", "python2.7",
+                                "site-packages", "h5py-*")) +
+            glob.glob(join_path(spec['py-numpy'].prefix, "lib", "python2.7",
+                                "site-packages", "numpy-*")))
+        try:
+            pythonpath.append(os.environ["PYTHONPATH"])
+        except KeyError:
+            pass
+        os.environ["PYTHONPATH"] = ":".join(pythonpath)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py]: I think this is all done automatically by anything that extends Python.

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    variant('julia', default=False)
+    variant('python', default=True)
+
+    variant('pic', default=True,
+            description="Produce position-independent code")
+
+    depends_on('julia', when='+julia', type=('build', 'run'))
+    depends_on('py-h5py', when='+python', type=('build', 'run'))
+    depends_on('py-numpy', when='+python', type=('build', 'run'))
+    depends_on('python@2.7.0:2.999.999', when='+python', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py]: Are you sure these should all be `build, run`? I've never seen someone link to Python or Julia with `build, run` before.

File: var/spack/repos/builtin/packages/simulationio/package.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import glob
+import os
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py]: Need one more line here to make flake8 happy.

File: var/spack/repos/builtin/packages/simulationio/package.py (hunk 3 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/simulationio/package.py]: Julia's packages call out Python, e.g. using `matplotlib` for visualization.

File: var/spack/repos/builtin/packages/simulationio/package.py (hunk 2 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/simulationio/package.py]: SimulationIO doesn't extend Python. Maybe it should. However, it is mostly a C++ library, it just also provides Python bindings. What's the best approach here?

File: var/spack/repos/builtin/packages/simulationio/package.py (hunk 3 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/simulationio/package.py]: Also, the build-time dependency is there to be able to run install these packages.

File: var/spack/repos/builtin/packages/simulationio/package.py (hunk 2 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/simulationio/package.py]: I use extends any time a build produces a `lib/python2.7/site-packages` directory. I think this should probably extend Python. It'll save you a lot of code. Extends is basically just depends_on but let's you activate the package and takes care of `PYTHONPATH` for you.

**Comment by eschnett** [comment]
I'm extending `python` now.

**Pull request merged**